mlflow/tracking/fluent.py

```diff
@@ -1342,32 +1332,12 @@ def _get_experiment_id_from_env():
 
 
 def _get_experiment_id():
-    # TODO: Replace with None for 1.0, leaving for 0.9.1 release backcompat with existing servers
-    deprecated_default_exp_id = "0"
-
     return (
         _active_experiment_id
         or _get_experiment_id_from_env()
```

Comment:
For completeness, can we introduce an `EnvironmentVariableExperimentProvider`?

Reply:
I don't see the in_context pattern here to use the provider pattern for this. Let me know if I am missing something.

Reply:
`"MLFLOW_EXPERIMENT_NAME" in os.environ`

Reply:
Discussed offline that we can't refactor this code based on how the code is currently implemented